File: app/src/main/java/com/elephantgroup/one/base/BaseFragment.java
package com.elephantgroup.one.base;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.elephantgroup.one.R;

import butterknife.ButterKnife;
import butterknife.Unbinder;

public abstract class BaseFragment<P extends BasePresenter> extends Fragment implements BaseView {

    private Unbinder mUnBinder;
    private View view;
    private boolean isDataFirstLoaded;
    protected TextView mTitle;
    protected ImageView mBack;


    @Override
    public void onCreate(Bundle savedInstanceState) {
        isDataFirstLoaded = true;
        super.onCreate(savedInstanceState);
    }


    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        if (!isDataFirstLoaded && view != null) {
            ((ViewGroup) view.getParent()).removeView(view);
            return view;
        }
        view = inflater.inflate(getLayoutId(), container, false);
        mUnBinder = ButterKnife.bind(this,view);
        initView();
        mPresenter = createPresenter();
        initData();
        return view;
    }

    private void initView() {
        mBack = view.findViewById(R.id.titleBack);
        mTitle = view.findViewById(R.id.titleName);
        if (mBack != null){
            mBack.setVisibility(View.GONE);
        }
    }

    protected void setTitle(String title) {
        if(mTitle != null){
            mTitle.setText(title);
        }
    }

    @SuppressLint("HandlerLeak")
    @Override
    public void onActivityCreated(Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        if (!isDataFirstLoaded) {
            return;
        }
        isDataFirstLoaded = false;
    }


    protected P mPresenter;

    public abstract int getLayoutId();

    public abstract P createPresenter();

    protected abstract void initData();

    @Override
    public void onDestroy() {
        super.onDestroy();
        mUnBinder.unbind();
    }
}

File: app/src/main/java/com/elephantgroup/one/base/BaseFragment.kt
package com.elephantgroup.one.base

import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import butterknife.ButterKnife
import butterknife.Unbinder
import com.elephantgroup.one.R

abstract class BaseFragment<P : BasePresenter> : Fragment(), BaseView {

    private var mUnBinder: Unbinder? = null
    protected var mTitle: TextView? = null
    protected var mBack: ImageView? = null
    protected var mPresenter: P? = null

    var tempView : View? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        var contentView = inflater.inflate(getLayoutId(), container, false)
        tempView = contentView;
        mUnBinder = ButterKnife.bind(this,contentView)
        initView()
        mPresenter = createPresenter()
        initData()
        if (contentView == null){
            return super.onCreateView(inflater, container, savedInstanceState)
        }
        return contentView
    }

    private fun initView() {
        mBack = tempView!!.findViewById(R.id.titleBack)
        mTitle = tempView!!.findViewById(R.id.titleName)
        if (mBack != null) {
            mBack!!.visibility = View.GONE
        }
    }

    protected fun setTitle(title: String) {
        if (mTitle != null) {
            mTitle!!.text = title
        }
    }

    abstract fun createPresenter(): P

    abstract fun getLayoutId(): Int

    protected abstract fun initData()

    override fun onDestroy() {
        super.onDestroy()
        mUnBinder!!.unbind()
    }
}